Simplified/Reader2/Bookmarks/NYPLBookmarkFactory.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  NYPLBookmarkFactory.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2022-09-13.
+//  Copyright © 2022 NYPL. All rights reserved.
+//
+
+import Foundation
+import R2Shared
+import NYPLUtilities
+import NYPLAudiobookToolkit
+
+class NYPLBookmarkFactory: NYPLBookmarkSelectorParsing {
+  /// Factory method to parse selector JSON value from a server annotation.
+  ///
+  /// - Parameters:
+  ///   - annotation: The annotation object coming from the server in a
+  ///   JSON-like structure.
+  ///   - annotationType: Whether it's an explicit bookmark or a reading progress.
+  ///   - bookID: The book the annotation is related to.
+  ///   missing from the annotation, e.g. if the annotation was created on an
+  ///   R1 client.
+  /// - Returns: a client-side representation of a bookmark.
+  class func parseSelectorJSONString(fromServerAnnotation annotation: [String: Any],
```

Comment:
The code in this function is extracted from `NYPLReadiumBookmarkFactory` class, since it can be used for both Readium and audiobook bookmarks.